### ISSUE TITLE
reproducing efficient net results

### ISSUE THREAD

**Issue by fgvbrt** [tensorflow/tpu#546]
Hello, I am trying to reproduce imagenet efficient net results on GPU  with standard data preprocessing. I followed this guide that have exact hyperparameters for b0 and b3 models: https://cloud.google.com/tpu/docs/tutorials/efficientnet 
I was able to reproduce results for b0;  for b3 took 50 more epochs to reproduce (about total 220 epochs). I tried to run b4 with same hyperparams as for b3, but I got 0.4% lower final top1 accuracy. 

I would like to ask few questions about training:
1) Can you please provide exact hyperparams for b4-b7 models (like in link above, as I understand you used different batch sizes for different models)?
2) I noticed that you are doing global batch norm with all reduce between differents shards, is it important for training big efficient model (I am currently using local batch norm, and do allreduce for mean/var and ema of mean var before scoring)?
3) I noticed that default option not to use bfloat16, am I right that you get worse results with bfloat16 or performance was close to fp32 training?

Thanks in advance!

**Comment by mingxingtan** [comment]
Hello Mikhail,

I used the same settings for all models for TPU training: total batch size 4096 and 350 epochs.  The only difference is that: for bigger models, we will train them with more TPU cores, so each core will has smaller batch size (but total batch size is always 4096).  Because each core has smaller batch size, we need to use global batch norm to maintain similar batch statistics for different models. Global norm is indeed important for large models (>=b3) because each core has smaller batch size.

Good question on bfloat16! It should have similar performance on TPUs, but I haven't tried it before. (If I understand correctly, bfloat16 may not work very well on GPUs).

Best,
Mingxing

**Comment by michaelklachko** [comment]
@fgvbrt Have you succeeded in training B4 from scratch without accuracy loss on GPUs? If so, can you provide your hyperparams please? How long did it take to train?

**Comment by steven-zzzz** [comment]
@mingxingtan Do you train all models **with AutoAugment/RandAugment** for 350 epochs too? Are there any differences in hyper-parameters when training B0-B7 with **AutoAugment/RandAugment**?
Thanks in advance!

**Comment by dwx5325834** [comment]
> @fgvbrt Have you succeeded in training B4 from scratch without accuracy loss on GPUs? If so, can you provide your hyperparams please? How long did it take to train?

Did you successfully reproduce it on the GPU?
